experimental/components/src/units/converter_factory.rs

```diff
@@ -0,0 +1,339 @@
+        ///     The sums are: meter: 1 + 1 = 2, second: -1 + (-1) = -2.
+        ///     If all the sums are zeros, then the units are reciprocal.
+        ///     If all the diffs are zeros, then the units are convertible.
+        ///     If none of the above, then the units are not convertible.
+        #[derive(Debug)]
+        struct PowersInfo {
+            diffs: i16,
+            sums: i16,
+        }
+
+        /// Inserting the units item into the map.
+        /// NOTE:
+        ///     This will require to go through the basic units of the given unit items.
+        ///     For example, `newton` has the basic units:  `gram`, `meter`, and `second` (each one has it is own power and si prefix).
+        fn insert_non_basic_units(
+            factory: &ConverterFactory,
+            units: &[MeasureUnitItem],
+            sign: i16,
+            map: &mut LiteMap<u16, PowersInfo>,
+        ) -> Option<()> {
+            for item in units {
+                let items_from_item = factory.payload.convert_infos.get(item.unit_id as usize);
+
+                debug_assert!(items_from_item.is_some(), "Failed to get convert info");
+
+                insert_base_units(items_from_item?.basic_units(), item.power as i16, sign, map);
+            }
+
+            Some(())
+        }
+
+        /// Inserting the basic units into the map.
+        /// NOTE:   
+        ///     The base units should be multiplied by the original power.
+        ///     For example, `square-foot` , the base unit is `meter` with power 1.
+        ///     Thus, the inserted power should be `1 * 2 = 2`.
+        fn insert_base_units(
+            basic_units: &ZeroSlice<MeasureUnitItem>,
+            original_power: i16,
+            sign: i16,
+            map: &mut LiteMap<u16, PowersInfo>,
+        ) {
+            for item in basic_units.iter() {
+                let item_power = (item.power as i16) * original_power;
+                let signed_item_power = item_power * sign;
+                if let Some(powers) = map.get_mut(&item.unit_id) {
+                    powers.diffs += signed_item_power;
+                    powers.sums += item_power;
+                } else {
+                    map.insert(
+                        item.unit_id,
+                        PowersInfo {
+                            diffs: (signed_item_power),
+                            sums: (item_power),
+                        },
+                    );
+                }
+            }
+        }
+
+        let unit1 = &unit1.contained_units;
+        let unit2 = &unit2.contained_units;
+
+        let mut map = LiteMap::new();
+        insert_non_basic_units(self, unit1, 1, &mut map)?;
+        insert_non_basic_units(self, unit2, -1, &mut map)?;
+
+        let (power_sums_are_zero, power_diffs_are_zero) =
+            map.iter_values()
+                .fold((true, true), |(sums, diffs), determine_convertibility| {
```

Comment:
nit: your struct is called `PowersInfo` now, but this argument is still called `determine_convertibility`

Reply:
done